sqld/src/replication/logger.rs

```diff
@@ -442,16 +605,45 @@ impl ReplicationLogger {
     }
 
     fn commit(&self, new_frame_count: u64, new_current_checksum: u64) {
-        let mut header = { *self.log_header.lock() };
+        let mut log_file = self.log_file.write();
+        let mut header = *log_file.header();
         header.frame_count = new_frame_count;
-
-        self.log_file
-            .write_all_at(bytes_of(&header), 0)
-            .expect("fatal error, failed to commit to log");
-
+        log_file.write_header(&header).expect("dailed to commit");
         self.current_checksum
             .store(new_current_checksum, Ordering::Relaxed);
-        *self.log_header.lock() = header;
+    }
+
+    pub async fn locate_snapshot(
+        &self,
+        start_idx: FrameId,
+    ) -> anyhow::Result<Option<SnapshotFile>> {
+        static SNAPSHOT_FILE_MATCHER: Lazy<Regex> = Lazy::new(
+            ||
+            //                                                   v---v-end idx
+            Regex::new(r#"(\w{8}-\w{4}-\w{4}-\w{4}-\w{12})-(\d*)-(\d*).snap"#).unwrap(),
```

Comment:
Tip: you can use the `x` regex flag (`(?x)` at the beginning) to ignore whitespace in the regex, which allows you to split the regex into multiple lines and include comments:
https://docs.rs/regex/latest/regex/#grouping-and-flags

Reply:
nice